src/http/one/FieldParser.cc

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 1996-2025 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+/* DEBUG: section 55    HTTP Header Field Parser */
+
+#include "squid.h"
+#include "base/CharacterSet.h"
+#include "base/Raw.h"
+#include "base/TextException.h"
+#include "debug/Stream.h"
+#include "http/one/FieldParser.h"
+#include "parser/Tokenizer.h"
+#include "sbuf/Stream.h"
+#include "SquidConfig.h"
+
+/**
+ * Governed by RFC 9112 section 5:
+ *
+ *  field-line   = field-name ":" OWS field-value OWS
+ */
+void
+Http1::FieldParser::parseFieldLine(::Parser::Tokenizer &tok, const http_hdr_owner_type msgType)
+{
+    // Isolate the: field-name
+    parseFieldName(tok);
+
+    // TODO: remove when String is gone from Squid
+    if (name().length() > 65534) {
+        /* String must be LESS THAN 64K and it adds a terminating NULL */
+        debugs(55, 2, "ignoring huge header field (" <<
+               Raw("field_start", name().rawContent(), 100) <<
+               "...[skip " << name().length()-100 << " characters])");
+        throw TextException(SBuf("huge-header-name"), Here());
+    }
+
+    // handle delimiter ( BWS ":" ) or abort
+    skipColonDelimiter(tok, msgType);
+
+    // Isolate the: OWS field-value OWS
+    (void)tok.skipAll(CharacterSet::WSP); // at least an SP expected
+    parseFieldValue(tok);
+    (void)tok.skipAll(CharacterSet::WSP);
+
+    // TODO: remove when String is gone from Squid
+    if (value().length() > 65534) {
+        /* String must be LESS THAN 64K and it adds a terminating NULL */
+        throw TextException(ToSBuf("'", name(), "' header of ", value().length(), " bytes"), Here());
+    }
```

Comment:
Please move this check into parseFieldValue() (instead of successfully parsing too-long values and only then checking their lengths). Similar for the name length check, of course.

Reply:
done in e222bc4134 along with matching check for field-name